core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -1468,23 +1472,23 @@ object Chunk extends CollectorK[Chunk] with ChunkCompanionPlatform {
   def concat[A](chunks: GSeq[Chunk[A]], totalSize: Int): Chunk[A] =
     if (totalSize == 0)
       Chunk.empty
-    else if (chunks.forall(c => c.knownElementType[Boolean] || c.forall(_.isInstanceOf[Boolean])))
+    else if (chunks.forall(c => c.knownElementType[Boolean] || c.isEmpty)) {
```

Comment:
The `.isInstanceOf` checks here are not safe on Scala.js -- a single number returns true for each numeric type it fits in.